crypto/evp/evp_fetch.c

```diff
@@ -112,30 +113,13 @@ static void *construct_method(const char *algorithm_name,
                               void *data)
 {
     struct method_data_st *methdata = data;
-    void *method = NULL;
-    int nid = OBJ_sn2nid(algorithm_name);
-
-    if (nid == NID_undef) {
-        /* Create a new NID for that name on the fly */
-        ASN1_OBJECT tmpobj;
-
-        /* This is the same as OBJ_create() but without requiring a OID */
-        tmpobj.nid = OBJ_new_nid(1);
-        tmpobj.sn = tmpobj.ln = methdata->name;
-        tmpobj.flags = ASN1_OBJECT_FLAG_DYNAMIC;
-        tmpobj.length = 0;
-        tmpobj.data = NULL;
-
-        nid = OBJ_add_object(&tmpobj);
-    }
+    int nid;
 
-    if (nid == NID_undef)
+    if ((nid = ossl_namemap_number(methdata->libctx, algorithm_name)) == 0
```

Comment:
The namemap_new seems to call ossl_namemap_number() and early out if its already mapped - so you may not need the first line.

Reply:
Things have changed since this comment...  care to take a refreshed look?